ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+TabWebDelegate.swift

```diff
@@ -0,0 +1,575 @@
+              }
+            }
+          } else {
+            self.addTab(url: url, inPrivateMode: true, currentTab: currentTab)
+          }
+        }
+        openNewPrivateTabAction.accessibilityLabel = "linkContextMenu.openInNewPrivateTab"
+
+        actions.append(openNewPrivateTabAction)
+
+        if UIApplication.shared.supportsMultipleScenes {
+          if !tabType.isPrivate {
+            let openNewWindowAction = UIAction(
+              title: Strings.openInNewWindowTitle,
+              image: UIImage(braveSystemNamed: "leo.window")
+            ) { _ in
+              self.openInNewWindow(url: url, isPrivate: false)
+            }
+
+            openNewWindowAction.accessibilityLabel = "linkContextMenu.openInNewWindow"
+            actions.append(openNewWindowAction)
+          }
+
+          let openNewPrivateWindowAction = UIAction(
+            title: Strings.openInNewPrivateWindowTitle,
+            image: UIImage(braveSystemNamed: "leo.window.tab-private")
+          ) { _ in
+            if !tabType.isPrivate, Preferences.Privacy.privateBrowsingLock.value {
+              self.askForLocalAuthentication { [weak self] success, error in
+                if success {
+                  self?.openInNewWindow(url: url, isPrivate: true)
+                }
+              }
+            } else {
+              self.openInNewWindow(url: url, isPrivate: true)
+            }
+          }
+
+          openNewPrivateWindowAction.accessibilityLabel = "linkContextMenu.openInNewPrivateWindow"
+          actions.append(openNewPrivateWindowAction)
+        }
+
+        let copyAction = UIAction(
+          title: Strings.copyLinkActionTitle,
+          image: UIImage(systemName: "doc.on.doc"),
+          handler: UIAction.deferredActionHandler { _ in
+            UIPasteboard.general.url = url as URL
+          }
+        )
+        copyAction.accessibilityLabel = "linkContextMenu.copyLink"
+        actions.append(copyAction)
+
+        let copyCleanLinkAction = UIAction(
+          title: Strings.copyCleanLink,
+          image: UIImage(braveSystemNamed: "leo.broom"),
+          handler: UIAction.deferredActionHandler { _ in
+            let service = URLSanitizerServiceFactory.get(privateMode: currentTab.isPrivate)
+            let cleanedURL = service?.sanitizeURL(url) ?? url
+            UIPasteboard.general.url = cleanedURL
+          }
+        )
+        copyCleanLinkAction.accessibilityLabel = "linkContextMenu.copyCleanLink"
+        actions.append(copyCleanLinkAction)
+
+        if let braveWebView = tab.webView {
+          let shareAction = UIAction(
+            title: Strings.shareLinkActionTitle,
+            image: UIImage(systemName: "square.and.arrow.up")
+          ) { _ in
+            let touchPoint = braveWebView.lastHitPoint
+            let touchRect = CGRect(origin: touchPoint, size: .zero)
+
+            // TODO: Find a way to add fixes #3323 and #2961 here:
```

Comment:
nit: can we prefix these with `brave-ios` while we are moving them (and below in same comment)? Ex. `brave-ios#3323`

Reply:
Will address in the next PR 🙂